### ISSUE TITLE
ValueError: unmergeable object classes 

### ISSUE THREAD

**Issue by aguzmanmesa** [lightkurve/lightkurve#954]
#### Problem description
Hi! 
I'm new to lightkurve. I was running this piece of code (below) and I haven't been able to work out the ValueError. I can't find similar errors within the documentation either. Any suggestions? Thank you in advance. 

#### Example

```python
import lightkurve as lk
from lightkurve import search_lightcurve
lc = lk.search_lightcurve("HD 93083", mission='TESS').download_all().stitch().remove_nans()
ValueError: unmergeable object classes ['Quantity', 'QColumn']

```

#### Environment

-  platform (e.g. Linux, OSX, Windows): Mac OSX 
-  lightkurve version (e.g. 1.0b6): 2.0.1
-  installation method (e.g. pip, conda, source): pip

**Comment by barentsen** [comment]
@aguzmanmesa Thank you for taking the time to open this issue! :pray:  I am able to reproduce this error.  It looks like a real bug we should fix.

**Note**: the two light curves you are trying to stitch together are both TESS Sector 9 light curves of the same object.  It doesn't make a lot of sense to stitch these together because you'd end up combining duplicate flux measurements into one light curve:

<img width="506" alt="Screen Shot 2021-02-18 at 7 56 13 PM" src="https://user-images.githubusercontent.com/817669/108455647-778bbb00-7223-11eb-94b0-e83797114ecd.png">

**Comment by barentsen** [comment]
It looks like this error is caused by the fact that we introduced a special `QColumn` class for reasons explained here: https://github.com/lightkurve/lightkurve/pull/891#issuecomment-718125631

I do not see a simple fix.  Once AstroPy v4.3 is released in a few months, the need for the `QColumn` class will go away and we could then fix this bug by removing the `QColumn` class.  This will be my preferred solution unless the bug affects many people.  Please comment below if it does!

**Comment by dfm** [comment]
I hit this issue too. Would perhaps be nice to get a more descriptive error, but I don't know much about what's happening in the background!

**Comment by barentsen** [comment]
Thank you for raising this issue @aguzmanmesa & @dfm!

In the example above, the underlying cause was that a SPOC and a QLP light curve were being stitched together.  The SAP_FLUX column in the former has units electron/second, whereas the latter is unitless/normalized.

I changed the behavior in #996 to exclude such incompatible columns from being stitched together, as opposed to having them raise the `ValueError` you encountered.  A warning is raised instead now:  

<img width="895" alt="Screen Shot 2021-03-12 at 1 03 54 PM" src="https://user-images.githubusercontent.com/817669/110998540-ccf34d80-8333-11eb-83b2-010bfb8b4576.png">

I suspect more problems related to stitching together light curves from different pipelines are hiding out there.  In general, it is probably best to limit the use of `stitch()` to light curves produced by the same pipeline, though I bet there are good use cases where breaking this advice makes sense!

**Comment by barentsen** [comment]
The fix for this has been merged and will be part of v2.0.5, expected some time next week.

Thanks again! 🙏
